lib/FrontendTool/FrontendTool.cpp

```diff
@@ -693,6 +693,42 @@ static void emitSwiftdepsForAllPrimaryInputsIfNeeded(
   }
 }
 
+static bool emitConstValuesForWholeModuleIfNeeded(
+    CompilerInstance &Instance) {
+  const auto &Invocation = Instance.getInvocation();
+  const auto &frontendOpts = Invocation.getFrontendOptions();
+  if (!frontendOpts.InputsAndOutputs.hasConstValuesOutputPath())
+    return false;
+  std::error_code EC;
+  assert(frontendOpts.InputsAndOutputs.isWholeModule() &&
+         "'emitConstValuesForWholeModule' only makes sense when the whole module can be seen");
+  auto ConstValuesFilePath = frontendOpts.InputsAndOutputs
+    .getPrimarySpecificPathsForAtMostOnePrimary().SupplementaryOutputs
+    .ConstValuesOutputPath;
+  llvm::raw_fd_ostream OS(ConstValuesFilePath, EC, llvm::sys::fs::OF_None);
+  OS << "{}\n";
+  return false;
+}
+
+static void emitConstValuesForAllPrimaryInputsIfNeeded(
+    CompilerInstance &Instance) {
+  const auto &Invocation = Instance.getInvocation();
+
+  for (auto *SF : Instance.getPrimarySourceFiles()) {
+    const std::string &ConstValuesFilePath =
+        Invocation.getConstValuesFilePathForPrimary(
+            SF->getFilename());
+    if (ConstValuesFilePath.empty()) {
+      continue;
+    }
+
+    // TODO: Emit extracted values
```

Comment:
I was wondering what that previous `OS << "{}\n";` was about before I saw this line. It makes sense now.